src/core/trtserver.cc

```diff
@@ -1156,6 +1203,26 @@ TRTSERVER_ServerRegisterSharedMemory(
   return nullptr;  // success
 }
 
+#ifdef TRTIS_ENABLE_GPU
+TRTSERVER_Error*
+TRTSERVER_ServerCudaRegisterSharedMemory(
```

Comment:
Why do we need a separate call? The TrtServerSharedMemoryBlock should know what type of shared memory it is and be able to do the right thing.

Reply:
Fixed